subworkflows/local/metagenomics.nf

```diff
@@ -0,0 +1,62 @@
+include { METAGENOMICS_COMPLEXITYFILTER } from './metagenomics_complexityfilter'
+include { METAGENOMICS_PROFILING        } from './metagenomics_profiling'
+include { METAGENOMICS_POSTPROCESSING   } from './metagenomics_postprocessing'
+
+workflow METAGENOMICS {
+    take: ch_bamfiltered_for_metagenomics
+
+    main:
+    // Define channels
+    ch_multiqc_files                = Channel.empty()
+    ch_versions                     = Channel.empty()
+
+    // Add single_end parameter to meta.
+    // Reads were merged before, so single_end is always true!
+    ch_bamfiltered_for_metagenomics = ch_bamfiltered_for_metagenomics.map{
+        meta, bamfiltered -> [meta+['single_end':true], bamfiltered]
+    }
+
+    //
+    // Run the complexity filter subworkflow
+    //
+
+    if ( params.run_metagenomics_complexityfiltering ) {
+        METAGENOMICS_COMPLEXITYFILTER( ch_bamfiltered_for_metagenomics )
+        ch_reads_for_metagenomics = METAGENOMICS_COMPLEXITYFILTER.out.fastq
+        ch_versions = ch_versions.mix(METAGENOMICS_COMPLEXITYFILTER.out.versions.first())
+        ch_multiqc_files = ch_multiqc_files.mix(METAGENOMICS_COMPLEXITYFILTER.out.fastq.collect{it[1]}.ifEmpty([]))
+    } else {
+        ch_reads_for_metagenomics = ch_bamfiltered_for_metagenomics
+    }
+
+    //
+    // Run the profiling subworkflow
+    //
+
+    database = Channel.fromPath(params.metagenomics_profiling_database)
```

Comment:
I would rather stage the files in the main script and pass to the metagenomics subworkflow, otherwise where they get iniitiallised (and possible validated) gets hidden away

Reply:
e.g. I would read this as as where are the NCBI taxdump files/resources for HOPS used?